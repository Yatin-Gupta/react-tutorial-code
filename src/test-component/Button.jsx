import React from "react";

function Button(props) {
  const { label } = props;
  return (
    <div>
      <button data-testid="button-testid">Hi {label}</button>
    </div>
  );
}

export default Button;
