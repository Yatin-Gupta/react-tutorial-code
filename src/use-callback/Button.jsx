import React from "react";

function Button(props) {
  const { children, clickHandler } = props;
  console.log("Rendering Button - " + children);
  return (
    <div>
      <button onClick={clickHandler}>{children}</button>
    </div>
  );
}

export default React.memo(Button);
