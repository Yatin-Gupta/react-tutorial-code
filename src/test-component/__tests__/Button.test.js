import React from "react";
import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";

import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Button from "../Button";

afterEach(cleanup);

it("render without any crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label={"Click Me"} />);
  expect(getByTestId("button-testid")).toHaveTextContent("Click Me");
});

it("renders another button correctly", () => {
  const { getByTestId } = render(<Button label={"Save Me"} />);
  expect(getByTestId("button-testid")).toHaveTextContent("Save Me");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(<Button label={"Save Snapshot"}></Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
