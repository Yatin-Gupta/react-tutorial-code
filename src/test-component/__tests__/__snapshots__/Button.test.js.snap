// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    data-testid="button-testid"
  >
    Hi 
    Save Snapshot
  </button>
</div>
`;
